server/controllers/shiftController.js

```diff
@@ -306,9 +327,14 @@ let ShiftController = function (
   };
 
   this.update = function (req, res) {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(422).json({message: "Invalid request", errors: errors.array()});
+    }
+
     // Check if user is admin
     if (!req.user.isAdmin) {
-      res.status(401).json({message: "Only admin can edit a shift"});
+      res.status(400).json({message: "Only admin can edit a shift"});
```

Comment:
Shouldn't this be 401?

Ditto in other cases.

Reply:
Yes.

Reply:
Done